test/MemeTest.sol

```diff
@@ -37,6 +38,36 @@ contract MemeTest is TestBaseContract {
         proxy.buyMeme(MemeBuySizeDollars.Five);
     }
 
+    function testZkSyncMockPurchaseMeme() public {
+        // deploy mock twap
+        MockTwap twap = new MockTwap();
+
+        twap.setSqrtPriceX96(1625 * 10 ** 18);
+        proxy.setPriceOracle(address(twap));
+
+        proxy.setAddress(LibConstants.TREASURY_ADDRESS, address(proxy));
+
+        console2.log(address(this).balance);
+
+        address user1 = address(0x1);
+
+        uint256 user1_startingBalance = user1.balance;
+
+        vm.prank(user1);
+        proxy.buyMeme{ value: 1 ether }(MemeBuySizeDollars.Five);
+        uint256 user1_endingBalance = user1.balance;
+
+        uint256 receiverBalance = address(proxy.getAddress(LibConstants.SERVER_ADDRESS)).balance;
+
+        console2.log("receiver balance", receiverBalance);
+        console2.log("user1 ending balance", user1_endingBalance);
+        assertEq(
+            user1_startingBalance,
+            user1_endingBalance + receiverBalance,
+            "user1 balance is not correct"
+        );
+    }
```

Comment:
I can't see where this test is checking the amount of MEME received by the user? What's the point of the test if it isn't checking that?

Reply:
fixed, will make the tests for meme more robust after merging